src/borg/remote.py

```diff
@@ -176,7 +180,17 @@ def __init__(self, location, create=False, lock_wait=None, lock=True, args=None)
         if version != RPC_PROTOCOL_VERSION:
             raise Exception('Server insisted on using unsupported protocol version %d' % version)
         try:
-            self.id = self.call('open', self.location.path, create, lock_wait, lock)
+            # Because of protocol versions, only send append_only if necessary
+            if append_only:
+                try:
+                    self.id = self.call('open', self.location.path, create, lock_wait, lock, append_only)
+                except self.RPCError as err:
+                    if err.remote_type == 'TypeError':
```

Comment:
could a TypeError happen under different circumstances than what we expect here?